File: app/src/main/java/net/emilla/util/Services.java
package net.emilla.util;

import android.app.AlarmManager;
import android.app.NotificationManager;
import android.content.ClipboardManager;
import android.content.Context;
import android.hardware.camera2.CameraManager;
import android.media.AudioManager;
import android.view.accessibility.AccessibilityManager;

public enum Services {
    ;

    public static AccessibilityManager accessibility(Context ctx) {
        return (AccessibilityManager) ctx.getSystemService(Context.ACCESSIBILITY_SERVICE);
    }

    public static AlarmManager alarm(Context ctx) {
        return (AlarmManager) ctx.getSystemService(Context.ALARM_SERVICE);
    }

    public static AudioManager audio(Context ctx) {
        return (AudioManager) ctx.getSystemService(Context.AUDIO_SERVICE);
    }

    public static CameraManager camera(Context ctx) {
        return (CameraManager) ctx.getSystemService(Context.CAMERA_SERVICE);
    }

    public static ClipboardManager clipboard(Context ctx) {
        return (ClipboardManager) ctx.getSystemService(Context.CLIPBOARD_SERVICE);
    }

    public static NotificationManager notification(Context ctx) {
        return (NotificationManager) ctx.getSystemService(Context.NOTIFICATION_SERVICE);
    }

}

File: app/src/main/java/net/emilla/util/Services.kt
@file:JvmName("Services")

package net.emilla.util

import android.app.AlarmManager
import android.app.NotificationManager
import android.content.ClipboardManager
import android.content.Context
import android.hardware.camera2.CameraManager
import android.media.AudioManager
import android.view.accessibility.AccessibilityManager

@JvmName("accessibility")
fun Context.accessibilityService(): AccessibilityManager {
    return getSystemService(Context.ACCESSIBILITY_SERVICE) as AccessibilityManager
}

@JvmName("alarm")
fun Context.alarmService(): AlarmManager {
    return getSystemService(Context.ALARM_SERVICE) as AlarmManager
}

@JvmName("audio")
fun Context.audioService(): AudioManager {
    return getSystemService(Context.AUDIO_SERVICE) as AudioManager
}

@JvmName("camera")
fun Context.cameraService(): CameraManager {
    return getSystemService(Context.CAMERA_SERVICE) as CameraManager
}

@JvmName("clipboard")
fun Context.clipboardService(): ClipboardManager {
    return getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager
}

@JvmName("notification")
fun Context.notificationService(): NotificationManager {
    return getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
}
